src/Components/Facility/Consultations/Feed.tsx

```diff
@@ -61,17 +56,14 @@ export const Feed: React.FC<IFeedProps> = ({ consultationId, facilityId }) => {
   const [statusReported, setStatusReported] = useState(false);
   const authUser = useAuthUser();
 
-  useEffect(() => {
-    const fetchFacility = async () => {
-      const res = await dispatch(getPermittedFacility(facilityId));
-
-      if (res.status === 200 && res.data) {
-        setCameraMiddlewareHostname(res.data.middleware_address);
+  useQuery(routes.getPermittedFacility, {
+    pathParams: { id: facilityId || "" },
+    onResponse: ({ res, data }) => {
+      if (res && res.status === 200 && data && data.middleware_address) {
+        setCameraMiddlewareHostname(data.middleware_address);
```

Comment:
You can directly do the following instead of having a useState for the middleware hostname.

```
const { data } = useQuery(...);

const cameraMiddlewareHostname = data?.middleware_address ?? "";
```